### PR TITLE
Updated dropdown cancel button style

### PR DESCRIPTION
**Pull request by bartaz**
Part of #686

As discussed there in comments we focus on updating button style only here, leaving layout updates to be confirmed and implemented later.

### CODE REVIEW

File: src/common/components/vanilla/button/button.css [hunk 1]
```diff
@@ -1,11 +1,4 @@
-$primary-button-color: $ubuntu-orange;
-$primary-button-color-hover: #d44615;
-$primary-button-text-color: #FFF;
-
-$secondary-button-color: #FFF;
```
Comment (bartaz) [inline src/common/components/vanilla/button/button.css]: Primary and secondary are not used anymore and are not part of Vanilla.

File: src/common/components/repository-row/dropdowns/register-name-dropdown.js [hunk 2]
```diff
@@ -231,9 +231,13 @@ const ActionButtons = (props) => {
...
     <div className={ styles.buttonRow }>
-      <a onClick={onCancelClick} className={ styles.cancel }>
+      <Button
+        appearance="base"
+        onClick={onCancelClick}
+      >
         Cancel
-      </a>
+      </Button>
+      {' '}
```
Comment (earnubs) [inline src/common/components/repository-row/dropdowns/register-name-dropdown.js]: This makes me think of the very complex button rules we had for button groups (in a pre Vanilla CSS lib), but you could obviously control this with `button + button:not(:last-child)` kind of thing.

File: src/common/components/vanilla/button/button.css [hunk 3]
```diff
@@ -44,53 +38,38 @@ $neutral-border-color: $dark-grey;
...
-  border: 1px solid $secondary-border-color;
 }
 
-.secondary:active,
-.secondary:focus,
-.secondary:hover {
-  background-color: $secondary-button-color-hover;
+.base:active,
+.base:focus,
+.base:hover {
```
Comment (earnubs) [inline src/common/components/vanilla/button/button.css]: In order: link, visited, hover, active?

Comment (bartaz) [inline src/common/components/vanilla/button/button.css]: I should dust out the "LoVe HAte" mnemonic ;)
I think it matters when there are different styles applied (in separate selectors).

**Pull request merged**